gtk4/src/subclass/text_view.rs

```diff
@@ -0,0 +1,424 @@
+    fn parent_paste_clipboard(&self, text_view: &Self::Type);
+    fn parent_set_anchor(&self, text_view: &Self::Type);
+    fn parent_snapshot_layer(
+        &self,
+        text_view: &Self::Type,
+        layer: TextViewLayer,
+        snapshot: Snapshot,
+    );
+    fn parent_toggle_overwrite(&self, text_view: &Self::Type);
+}
+
+impl<T: TextViewImpl> TextViewImplExt for T {
+    fn parent_backspace(&self, text_view: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkTextViewClass;
+            if let Some(f) = (*parent_class).backspace {
+                f(text_view.unsafe_cast_ref::<TextView>().to_glib_none().0)
+            }
+        }
+    }
+
+    fn parent_copy_clipboard(&self, text_view: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkTextViewClass;
+            if let Some(f) = (*parent_class).copy_clipboard {
+                f(text_view.unsafe_cast_ref::<TextView>().to_glib_none().0)
+            }
+        }
+    }
+
+    fn parent_cut_clipboard(&self, text_view: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkTextViewClass;
+            if let Some(f) = (*parent_class).cut_clipboard {
+                f(text_view.unsafe_cast_ref::<TextView>().to_glib_none().0)
+            }
+        }
+    }
+
+    fn parent_delete_from_cursor(&self, text_view: &Self::Type, type_: DeleteType, count: i32) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkTextViewClass;
+            if let Some(f) = (*parent_class).delete_from_cursor {
+                f(
+                    text_view.unsafe_cast_ref::<TextView>().to_glib_none().0,
+                    type_.to_glib(),
+                    count,
+                )
+            }
+        }
+    }
+
+    fn parent_extend_selection(
+        &self,
+        text_view: &Self::Type,
+        granularity: TextExtendSelection,
+        location: &TextIter,
+        start: &mut TextIter,
+        end: &mut TextIter,
+    ) -> glib::signal::Inhibit {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkTextViewClass;
+            if let Some(f) = (*parent_class).extend_selection {
+                glib::signal::Inhibit(from_glib(f(
+                    text_view.unsafe_cast_ref::<TextView>().to_glib_none().0,
+                    granularity.to_glib(),
+                    location.to_glib_none().0,
+                    start.to_glib_none_mut().0,
+                    end.to_glib_none_mut().0,
+                )))
+            } else {
+                glib::signal::Inhibit(false)
+            }
+        }
+    }
```

Comment:
I think this should return glib::Inhibit but not very sure if I've understood that correctly from the C implementation. The corresponding C code is at https://gitlab.gnome.org/GNOME/gtk/-/blob/master/gtk/gtktextview.c#L7070

Reply:
In GTK4 this is fortunately a bit clearer:

> GDK_EVENT_STOP to stop other handlers from being invoked for the event. GDK_EVENT_PROPAGATE to propagate the event further.

That's exactly `Inhibit`. You can map to `glib::Inhibit` here or alternatively also create a new enum in gdk4 for that.

And you'll want to also properly map such bools to `Inhibit` for signal handlers in `src/auto` etc :)

Reply:
I guess something like https://github.com/gtk-rs/gtk-rs/issues/91 would make sense here as well

Reply:
I will keep using glib::Inhibit for now here and opened https://github.com/gtk-rs/gtk4-rs/issues/88 to track and fix that issue properly

Reply:
Makes sense, yes